stempy/electron.cpp

```diff
@@ -123,14 +128,17 @@ inline uint16_t mod(uint16_t x, uint16_t y)
 
 // Return the points in the frame with values larger than all 8 of their nearest
 // neighbors
-std::vector<uint32_t> maximalPoints(
-  const std::vector<uint16_t>& frame, int width, int height)
+std::vector<uint32_t> maximalPoints(const std::vector<uint16_t>& frame,
+                                    Dimensions2D frameDimensions)
 {
+  auto width = frameDimensions.first;
+  auto height = frameDimensions.second;
+
   std::vector<uint32_t> events;
   auto numberOfPixels = height * width;
   for (int i = 0; i < numberOfPixels; i++) {
-    auto row = i / width;
-    auto column = i % width;
+    int row = i / width;
+    int column = i % width;
```

Comment:
Why `int` here?

Reply:
Similar to [here](https://github.com/OpenChemistry/stempy/pull/109/files/e54292bdde442e913199e200b21bfc2dd25cc4e4#diff-93e8605537e636aa2f094e4472e46b24R26), I made `row` an int so that `row - 1` (see below) won't do an unsigned integer underflow. I'm **think** `row` can be 0. `column` doesn't need to be an `int`, but I just did it for consistency...

Reply:
@cjh1 and I discussed this, and we realized that we need to make sure the pixels on the borders don't try to check neighbors outside the image (like, currently, the pixels on the top border are trying to compare pixels above them). Once we change that, we will not need to make the row and column into `int` anymore either.

Reply:
This is fixed.